include/oneapi/dpl/pstl/hetero/algorithm_impl_hetero.h

```diff
@@ -363,12 +359,7 @@ _ForwardIterator
 __pattern_generate_n(__hetero_tag<_BackendTag> __tag, _ExecutionPolicy&& __exec, _ForwardIterator __first,
                      _Size __count, _Generator __g)
 {
-    // TODO: is this new implementation are correct?
-    // Previously we hadn't hetero impl for __pattern_generate_n
-
-    return __pattern_walk1_n(__tag, ::std::forward<_ExecutionPolicy>(__exec),
-                             __par_backend_hetero::make_iter_mode<__par_backend_hetero::access_mode::write>(__first),
-                             __count, generate_functor<_Generator>{__g});
+    return __pattern_generate(__tag, ::std::forward<_ExecutionPolicy>(__exec), __first, __first + __count, __g);
```

Comment:
`__internal::__pattern_generate`?

Reply:
Why do you need to explicitly specify the namespace `__internal` ?
We already in this namespace. And we have a lot of calls like this (without specify namespace `__internal`) for other patterns (~32).

Reply:
Will change later in separate PR